README.md

```diff
@@ -118,71 +135,79 @@ vault write data/buildkite/my_pipeline/git-credentials value=- <<< $(echo "https
 These are then exposed via a [gitcredential helper](https://git-scm.com/docs/gitcredentials) which will download the
 credentials as needed.
 
-### Environment variables
+## Options
+***
+The Vault Secrets plugin supports a number of different configuration options.
 
-Key values pairs can also be uploaded.
+### `server`
+The address of the target Vault server. Example: `https://my-vault-server:8200`
 
-```bash
-vault write data/buildkite/my_pipeline/environment value=- <<< $(echo "MY_SECRET=blah" | base64)
-```
+### `path`
+Alternative Base Path to use for Vault secrets. This is expected to be a [KV Store](https://developer.hashicorp.com/vault/docs/secrets/kv#kv-version-2)  
 
-```bash
-vault write data/buildkite/my_pipeline/env_key value=- <<< $(echo "my secret"| base64)
-```
+Defaults to: `data/buildkite`
 
-Can be loaded using:
+
+### `auth`
+Configure the authentication parameters the plugin should use to authenticate with Vault.
+
+`auth` accepts the following options:
+
+`method` (required)
+The auth method to use when authenticating with Vault. Currently only `approle` is supported
+
+`approle`
+Configures the plugin to use AppRole authentication to the Vault server. Requires a `role-id` be set.
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`role-id` (required)
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; The role-id the plugin should use to authenticate to Vault
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`secret-env`
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; The environment variable which holds the **secret-id** used to authenticate to vault. Default `VAULT_SECRET_ID`
+
+
```

Comment:
As mentioned before, I would change the format of these keys to a similar one used for all other top-level plugin options:

```suggestion
#### `method` (required)

The auth method to use when authenticating with Vault. Currently only `approle` is supported

Possible values:
* `approle`: use AppRole authentication to the Vault server (requires a `role-id` be set)

#### `role-id` (required for `approle`)

The role-id the plugin should use to authenticate to Vault. Has no default value

#### `secret-env` (optional)

The environment variable which holds the **secret-id** used to authenticate to vault. Default `VAULT_SECRET_ID`
```